sdk/core/Azure.Core/src/Shared/DictionaryHeaders.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+#nullable disable
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// An implementation for manipulating headers on <see cref="Request"/>.
+    /// </summary>
+    public class DictionaryHeaders
+    {
+        private readonly Dictionary<string, object> _headers = new(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Initializes an instance of <see cref="DictionaryHeaders"/>
+        /// </summary>
+        public DictionaryHeaders()
+        { }
+
+        /// <summary>
+        /// Adds a header value to the header collection.
+        /// </summary>
+        /// <param name="name">The header name.</param>
+        /// <param name="value">The header value.</param>
+        public void AddHeader(string name, string value)
+        {
+            if (!_headers.TryGetValue(name, out object objValue))
+            {
+                _headers[name] = value;
+            }
+            else
+            {
+                if (objValue is not List<string> values)
```

Comment:
```suggestion
                if (objValue is List<string> values)
```

and reverse the if.

Reply:
I'm not sure if that is correct. This line checks to see if the existing value is not a list (a string) and therefore needs to be upgraded to a List initialized with the existing value and the new value.

Reply:
we know the value is not null so it's either a string or a `List`. So the following should be equivalent.

``` C#
if (objValue is List<string> values)
{
   values.Add(value);
}
else
{
    _headers[name] = new List<string> { objValue as string, value };
}
```
Or do I miss something subtle?

Reply:
Yes, your snippet is equivalent to what is there now. Perhaps I am missing why you are recommending reversing them.

Reply:
My bad, I was just commenting that without the negative in the condition it might read easier.